pkg/apis/apps/v1alpha1/apimanagerbackup_types.go

```diff
@@ -0,0 +1,156 @@
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+
+// APIManagerBackupSpec defines the desired state of APIManagerBackup
+// +k8s:openapi-gen=true
+type APIManagerBackupSpec struct {
+	// Backup data destination configuration
+	BackupDestination APIManagerBackupDestination `json:"backupDestination"`
+}
+
+// APIManagerBackupDestination defines the backup data destination
+// configurability. It is a union type. Only one of the fields can be
+// set
+type APIManagerBackupDestination struct {
+	// PersistentVolumeClaim as backup data destination configuration
+	// +optional
+	PersistentVolumeClaim *PersistentVolumeClaimBackupDestination `json:"persistentVolumeClaim,omitempty"`
+}
+
+// PersistentVolumeClaimBackupDestination defines the configuration
+// of the PersistentVolumeClaim to be used as the backup data destination
+// Ways to define a PVC creation:
+// Define VolumeName OR Define Resources. When VolumeName is specified resources is not needed:
+// Detailed information:
+// https://docs.okd.io/3.11/dev_guide/persistent_volumes.html#persistent-volumes-volumes-and-claim-prebinding
+type PersistentVolumeClaimBackupDestination struct {
+	// Resources configuration for the backup data PersistentVolumeClaim.
+	// Ignored when VolumeName field is set
+	// +optional
+	Resources *PersistentVolumeClaimResources `json:"resources,omitempty"`
+	// Name of an existing PersistentVolume to be bound to the
+	// backup data PersistentVolumeClaim
+	// +optional
+	VolumeName *string `json:"volumeName,omitempty"`
+	// Storage class to be used by the PersistentVolumeClaim. Ignored
+	// when VolumeName field is set
+	// +optional
+	StorageClass *string `json:"storageClass,omitempty"`
+}
+
+// PersistentVolumeClaimResources defines the resources configuration
+// of the backup data destination PersistentVolumeClaim
+type PersistentVolumeClaimResources struct {
+	// Storage Resource requests to be used on the PersistentVolumeClaim.
+	// To learn more about resource requests see:
+	// https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	Requests resource.Quantity `json:"requests"` // Should this be a string or a resoure.Quantity? it seems it is serialized as a string
+}
+
+type APIManagerBackupConditionType string
+
+// APIManagerBackupCondition describes the state of an APIManagerBackup at a certain point.
+type APIManagerBackupCondition struct {
```

Comment:
I have used this same type for other features:

https://github.com/3scale/3scale-operator/blob/capabilities-v2/pkg/common/status_conditions.go#L57

It does not have anything specific for backup or restore

Reply:
:+1:  operator-sdk is also adding that in future versions. If needed that could be used when upgrading